shared/js/search_provider.js

```diff
@@ -30,32 +32,65 @@
   // Allow consumers to wait for data to be initialised
   var readyPromise;
 
-  // TODO: This should implement how we pick the partner from
-  // the details of the users build
-  function getPartnerCode(cb) {
-    cb('default');
+  function extend(dest, source) {
+    for (var k in source) {
+      if (source.hasOwnProperty(k)) {
+        var value = source[k];
+        if (dest.hasOwnProperty(k) &&
+            typeof dest[k] === 'object' &&
+            typeof value === 'object') {
+          extend(dest[k], value);
+        } else {
+          dest[k] = value;
+        }
+      }
+    }
+    return dest;
+  }
+
+  function normalize(aCode) {
```

Comment:
Can we use this utility instead: https://github.com/mozilla-b2g/gaia/blob/master/shared/js/format.js#L16